### PR TITLE
Fix complex warnings

### LINKED ISSUE #1212: [BUG] `ComplexWarning`s in various demos
#### Issue description

Many demos have complex warnings present, with some demos even having multiple complex warnings present.

**Expected behavior:** (What you expect to happen)
No warnings are expected

**Actual behavior:** (What actually happens)
The following demos have `ComplexWarning`s:
* [`tutorial_eqnn_force_field`](https://pennylane.ai/qml/demos/tutorial_eqnn_force_field/): 1 warning
* [`tutorial_here_comes_the_sun`](https://pennylane.ai/qml/demos/tutorial_here_comes_the_sun/): 1 warning
* [`tutorial_pulse_programming101`](https://pennylane.ai/qml/demos/tutorial_pulse_programming101/): 4 warnings
* [`tutorial_vqe_vqd`](https://pennylane.ai/qml/demos/tutorial_vqe_vqd/): 34 warnings (yikes)

**System information:** (include your operating system, Python version, Sphinx version, etc.)

#### Source code and tracebacks

The warnings in all demos is from jax. Example warning:
```pycon
/home/runner/work/qml/qml/venv/lib/python3.10/site-packages/jax/_src/lax/lax.py:2745: ComplexWarning: Casting complex values to real discards the imaginary part
  x_bar = _convert_element_type(x_bar, x.aval.dtype, x.aval.weak_type)
```

### PR DESCRIPTION
**Pull request by mudit2812**
As name says.

* Updated `tutorial_vqe_vqd` to cast `h2.hf_state` as a numpy array. It's a pennylane numpy array by default, which was causing warnings about mixing autograd and jax interfaces.
* Suppress `ComplexWarning`s in `tutorial_vqe_vqd`.
* Cast `data` in `tutorial_eqnn_force_field` as complex, since we're `einsum`-ing it with a complex tensor.

These changes will not fix all `ComplexWarning`s. That will require changes in pennylane (see [here](https://github.com/PennyLaneAI/pennylane/pull/6256)).

**NOTE**
After checking all build artifacts locally, I can confirm that the following `ComplexWarning`s remain:

* 4 warnings in `tutorial_pulse_programming`
* 1 warning in `tutorial_here_comes_the_sun`

Partially fixes #1212 

[sc-73429]

### CODE REVIEW

**Comment by mudit2812** [comment]
Once `v0.38.1` is released, I will run CI in this PR with the `build-all-demos` label and add filter all `ComplexWarning`s to be errors.

File: demonstrations/tutorial_vqe_vqd.py [hunk 1]
```diff
@@ -45,9 +45,17 @@
 # To implement the VQD algorithm, we first need to know the ground state of our system, and it is a breeze to use the data from `PennyLane Datasets <https://pennylane.ai/datasets/>`__  to obtain the Hamiltonian and the ground state
 # of the hydrogen molecule:
 #
+# .. note::
+#
+#     To improve veiwability of this tutorial, we will suppress any ``ComplexWarning``'s which may raised during optmization.
```
Comment (dwierichs) [inline demonstrations/tutorial_vqe_vqd.py:50]:
```suggestion
#     To improve viewability of this tutorial, we will suppress any ``ComplexWarning``'s which may be raised during optimization.
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 2]
```diff
@@ -317,7 +317,7 @@ def vqlm(data, params):
 
     # Initial encoding
     for i in range(num_qubits):
-        equivariant_encoding(alphas[i, 0], jnp.asarray(data)[i % active_atoms, ...], wires=[i])
+        equivariant_encoding(alphas[i, 0], jnp.asarray(data, dtype=complex)[i % active_atoms, ...], wires=[i])
```
Comment (dwierichs) [inline demonstrations/tutorial_eqnn_force_field.py:320]: These two changes lead to horizontal scrollbars, unfortunately...

```suggestion
        equivariant_encoding(
            alphas[i, 0], jnp.asarray(data, dtype=complex)[i % active_atoms, ...], wires=[i]
        )
```

File: demonstrations/tutorial_eqnn_force_field.py [hunk 3]
```diff
@@ -339,7 +339,7 @@ def vqlm(data, params):
         # Encoding
         for i in range(num_qubits):
             equivariant_encoding(
-                alphas[i, d + 1], jnp.asarray(data)[i % active_atoms, ...], wires=[i]
+                alphas[i, d + 1], jnp.asarray(data, dtype=complex)[i % active_atoms, ...], wires=[i]
```
Comment (dwierichs) [inline demonstrations/tutorial_eqnn_force_field.py:342]:
```suggestion
                alphas[i, d + 1],
                jnp.asarray(data, dtype=complex)[i % active_atoms, ...],
                wires=[i],
```

**Review by dwierichs — CHANGES_REQUESTED** [review]
Had 3 small suggestions.

There seems to be a problem with the deployed version, not sure what that is?

Otherwise lgtm!

**Comment by mudit2812** [comment]
> There seems to be a problem with the deployed version, not sure what that is?

Yeah I'm not sure what that's about. If I manually try to open the deployment I'm seeing the most recent changes, so I'm confused. Regardless, after committing your suggestions, hopefully the next deployment will succeed.

**Review by dwierichs — APPROVED** [review]
:rocket:

**Comment by mudit2812** [comment]
Looks like the deployment is successful, but the action showed as failed because the step for adding/updating the comment about the deployment on the PR fails.See [here](https://github.com/PennyLaneAI/qml/actions/runs/10890656245/job/30219849910).

**Comment by mudit2812** [comment]
Since I've verified that all demos build and no new warnings are introduced, I'm removing the build-all-demos label.

**Review by soranjh — APPROVED** [review]

File: demonstrations/tutorial_vqe_vqd.py [hunk 4]
```diff
@@ -45,9 +45,17 @@
 # To implement the VQD algorithm, we first need to know the ground state of our system, and it is a breeze to use the data from `PennyLane Datasets <https://pennylane.ai/datasets/>`__  to obtain the Hamiltonian and the ground state
 # of the hydrogen molecule:
 #
+# .. note::
+#
+#     To improve viewability of this tutorial, we will suppress any ``ComplexWarning``'s which may be raised during optimization.
```
[the remark this answers is not in the document]
Comment (mudit2812) [reply demonstrations/tutorial_vqe_vqd.py:50]: Sphinx fails without the tick. You can't have an alphabet immediately after inline code or latex.
